### ISSUE TITLE
fwaas_v2: doesn't extract HTTP headers

### ISSUE THREAD

**Issue by kayrus** [gophercloud/gophercloud#3184]
```
$ egrep -rl ', r.Err = c\.[A-Z][a-z]+\('
openstack/networking/v2/extensions/fwaas_v2/groups/requests.go
openstack/networking/v2/extensions/fwaas_v2/policies/requests.go
```

The code was added in #2050 after the #1942 was merged.

Maybe we can implement a linter to avoid such cases?

**Comment by pierreprinetti** [comment]
Can you please explain what you are grepping and why, and give me a counter-example of how it should be done?

**Comment by kayrus** [comment]
see #1942 to get more context ;)

**Comment by kayrus** [comment]
@pierreprinetti apologies for not being clear enough. There were a couple of related fixes that I recall: #1942 #1952 and I mixed them up, my bad. Long story short, we must follow a pattern and call `_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)` after each HTTP request, otherwise new code can copy the wrong pattern like it happened in #2050. I wonder whether it's possible to create a linter to catch such cases.

**Comment by mandre** [comment]
It looks like we should be looking at any call to `Get`, `Post`, etc that ignores the HTTP response.

The following command yields one more result:

```
$ ag "_,.+\.(Get|Post|Put|Patch|Delete|Head)\(" openstack --ignore "*_test.go"
openstack/networking/v2/extensions/fwaas_v2/groups/requests.go
72:     _, r.Err = c.Get(ctx, resourceURL(c, id), &r.Body, nil)
110:    _, r.Err = c.Post(ctx, rootURL(c), b, &r.Body, nil)
145:    _, r.Err = c.Put(ctx, resourceURL(c, id), b, &r.Body, &gophercloud.RequestOpts{
168:    _, r.Err = c.Put(ctx, resourceURL(c, id), b, &r.Body, &gophercloud.RequestOpts{
184:    _, r.Err = c.Put(ctx, resourceURL(c, id), b, &r.Body, &gophercloud.RequestOpts{
192:    _, r.Err = c.Delete(ctx, resourceURL(c, id), nil)

openstack/networking/v2/extensions/fwaas_v2/policies/requests.go
94:     _, r.Err = c.Post(ctx, rootURL(c), b, &r.Body, nil)
100:    _, r.Err = c.Get(ctx, resourceURL(c, id), &r.Body, nil)
133:    _, r.Err = c.Put(ctx, resourceURL(c, id), b, &r.Body, &gophercloud.RequestOpts{
141:    _, r.Err = c.Delete(ctx, resourceURL(c, id), nil)
165:    _, r.Err = c.Put(ctx, insertURL(c, id), b, &r.Body, &gophercloud.RequestOpts{
173:    _, r.Err = c.Put(ctx, removeURL(c, id), b, &r.Body, &gophercloud.RequestOpts{

openstack/utils/choose_version.go
141:    _, err := client.Get(ctx, client.Endpoint, &resp, &gophercloud.RequestOpts{
```
